internal/indexer/indexer.go

```diff
@@ -854,6 +866,497 @@ func (i *Indexer) indexReferenceToExternalDefinition(p *packages.Package, docume
 	return rangeID, true
 }
 
+type def struct {
+	pkg   *packages.Package
+	obj   types.Object
+	ident *ast.Ident
+}
+
+func extractTypes(pkgs []*packages.Package) ([]def, []def) {
+	interfaces := []def{}
+	concreteTypes := []def{}
+	for _, pkg := range pkgs {
+		for ident, obj := range pkg.TypesInfo.Defs {
+			if obj == nil {
+				continue
+			}
+
+			// We ignore aliases 'type M = N' to avoid duplicate reporting
+			// of the Named type N.
+			if _, ok := obj.(*types.TypeName); !ok {
+				continue
+			}
+
+			if _, ok := obj.Type().(*types.Named); !ok {
+				continue
+			}
+
+			// methodLen := types.NewMethodSet(obj.Type()).Len()
+			// methodLen := listMethods(obj.Type())
+			methodLen := len(listMethods(obj.Type().(*types.Named)))
+			if methodLen == 0 {
+				// fmt.Println("OH NO NO NO", obj.Name())
+				continue
+			}
+
+			if types.IsInterface(obj.Type()) {
+				// TODO figure out non-exported interfaces
+				// should link within package? across packages?
+				interfaces = append(interfaces, def{pkg: pkg, obj: obj, ident: ident})
+			} else {
+				concreteTypes = append(concreteTypes, def{pkg: pkg, obj: obj, ident: ident})
+			}
+		}
+	}
+
+	return interfaces, concreteTypes
+}
+
+// indexImplementations emits data for each implementation of an interface.
+func (i *Indexer) indexImplementations() error {
+	// Load all dependencies
+	deps, err := i.loadDependencyPackages()
+	if err != nil {
+		return err
+	}
+
+	start1 := time.Now()
+	localInterfaces, localConcreteTypes := extractTypes(i.packages)
+	remoteInterfaces, remoteConcreteTypes := extractTypes(deps)
```

Comment:
i'd still be somewhat interested in us seeing if we can calculate the implemetations on a per-dependency basis (and then parallelize that via go funcs) and see if we get any noticeable performance improvements on large repos.